src/script/cpp_api/s_player.cpp

```diff
@@ -179,6 +179,36 @@ void ScriptApiPlayer::on_cheat(ServerActiveObject *player,
 	runCallbacks(2, RUN_CALLBACKS_MODE_FIRST);
 }
 
+void ScriptApiPlayer::on_moveplayer(ServerActiveObject* player, 
+		const v3s16 newpos, const v3s16 oldpos)
+{
+	SCRIPTAPI_PRECHECKHEADER
+
+	// Get core.registered_on_moveplayer
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "registered_on_moveplayer");
+	// Call callbacks
+	objectrefGetOrCreate(L, player);
+	
+	lua_newtable(L);
+	lua_pushnumber(L, newpos.X);
```

Comment:
if i'm correct we already have helpers to write v3f or v3s16 to lua

Reply:
I think so too, I got this version from here https://github.com/minetest/minetest/blob/9649e4721467dab348011633c814a63a184bd018/src/script/lua_api/l_object.cpp#L118

Reply:
https://github.com/minetest/minetest/blob/4118e150f8841a507e3fcb546902d1e07150e4a2/src/script/common/c_converter.cpp#L290

`push_v3s16` does that for you.